lms/djangoapps/django_comment_client/forum/tests.py

```diff
@@ -531,9 +531,133 @@ def test_group_info_in_ajax_response(self, mock_request):
         )
 
 
+@patch('requests.request')
+class SingleThreadContentGroupTestCase(ContentGroupTestCase):
+    def assert_can_access(self, user, discussion_id, thread_id, should_have_access):
+        """
+        Verify that a user has access to a thread within a given
+        discussion_id when should_have_access is True, otherwise
+        verify that the user does not have access to that thread.
+        """
+        request = RequestFactory().get("dummy_url")
+        request.user = user
+        mako_middleware_process_request(request)
+
+        def call_single_thread():
+            return views.single_thread(
+                request,
+                unicode(self.course.id),
+                discussion_id,
+                thread_id
+            )
+
+        if should_have_access:
+            self.assertEqual(call_single_thread().status_code, 200)
+        else:
+            with self.assertRaises(Http404):
+                call_single_thread()
+
+    def assert_searched_with_discussion_ids(self, mock_request, expected_commentable_ids):
+        """
+        Verify that the comments service was searched for threads with
+        the expected discussion ids (passed to the comments service as
+        'commentable_ids').
+        """
+        mock_request.assert_called_with(
+            'get',
+            StringEndsWithMatcher('threads'),
+            headers=ANY,
+            timeout=ANY,
+            data=None,
+            params=PartialDictMatcher({
```

Comment:
So don't we instead want to check that only the expected things are present in some_dict? Otherwise, this test could falsely pass if the code is passing things that it should not.